linux_testing_bcachefs: Bump patch version

DarkKirb opened the pull request:
Currently, bcachefs in nixpkgs can panic in certain cases, when the file system has subvolumes (or snapshots, which are based on subvolumes).

In particular, I have found that installing nixos caused this panic to happen reliably.

More information about the panic in question:
https://lore.kernel.org/all/bc622d24-9fad-7b3-22cb-da4bf2dd32d@ewheeler.net/T/

This issue has already been fixed upstream

###### Description of changes

- The version of the linux_testing_bcachefs package was bumped to the latest revision when I created this patch (c38b7167aa5f3b1b91dcc93ade57f30e95064590)
- linux_testing_bcachefs is based on 5.16 now, as upstream moved to that branch
- As per the comment in pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix, bcachefs-tools was updated to the latest version as well.

Reading through the commit titles, it appears that the new changes are mostly various bugfixes.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable: Installed and booted a NixOS system with the new patch version
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Madouura commented:
LGTM, just a couple extra things I think need to be changed.
- Difference hashes are in a different format, please use the same format as the previous commits.
- Please split up bcachefs-tools and bcachefs kernel into separate commits, linux-testing-bcachefs.nix linux-kernels.nix should be in the same commit.
Here's a [patch](https://github.com/NixOS/nixpkgs/files/8352158/fixup.patch.txt) you can use.
```diff
diff --git a/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix b/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix
index 63d47dd8a38..e86f6fced56 100644
--- a/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix
+++ b/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix
@@ -1,9 +1,9 @@
 { lib
 , fetchpatch
 , kernel
-, date ? "2022-03-09"
-, commit ? "2280551cebc1735f74eef75d650dd5e175461657"
-, diffHash ? "1mrrydidbapdq0fs0vpqhs88k6ghdrvmjpk2zi7xlwj7j32h0nwp"
+, date ? "2022-03-21"
+, commit ? "c38b7167aa5f3b1b91dcc93ade57f30e95064590"
+, diffHash ? "04lgwnng7p2rlz9sxn74n22750kh524xwfws3agqs12pcrvfsm0j"
 , kernelPatches # must always be defined in bcachefs' all-packages.nix entry because it's also a top-level attribute supplied by callPackage
 , argsOverride ? {}
 , ...
diff --git a/pkgs/tools/filesystems/bcachefs-tools/default.nix b/pkgs/tools/filesystems/bcachefs-tools/default.nix
index aa049682b83..e5325b1f988 100644
--- a/pkgs/tools/filesystems/bcachefs-tools/default.nix
+++ b/pkgs/tools/filesystems/bcachefs-tools/default.nix
@@ -22,13 +22,13 @@
 
 stdenv.mkDerivation {
   pname = "bcachefs-tools";
-  version = "unstable-2022-03-09";
+  version = "unstable-2022-03-22";
 
   src = fetchFromGitHub {
     owner = "koverstreet";
     repo = "bcachefs-tools";
-    rev = "3e2e3d468eed1d5ebbb4c6309d2eaebd081912c5";
-    sha256 = "1sb0dj2whlp3dxgf642z7yx7s8va5ah82zi6r4qni7l64qy1n554";
+    rev = "f3cdace86c8b60a4efaced23b2d31c16dc610da9";
+    sha256 = "1hg4cjrs4yr0mx3mmm1jls93w1skpq5wzp2dzx9rq4w5il2xmx19";
   };
 
   postPatch = ''
diff --git a/pkgs/top-level/linux-kernels.nix b/pkgs/top-level/linux-kernels.nix
index e73b5ebda1d..bd7bacf3000 100644
--- a/pkgs/top-level/linux-kernels.nix
+++ b/pkgs/top-level/linux-kernels.nix
@@ -186,7 +186,7 @@ in {
        else testing;
 
     linux_testing_bcachefs = callPackage ../os-specific/linux/kernel/linux-testing-bcachefs.nix rec {
-      kernel = linux_5_15;
+      kernel = linux_5_16;
       kernelPatches = kernel.kernelPatches;
    };
 
```

Thankfully, the on-disk format change seems to be automatic, so we should be able to backport this as well.

Madouura reviewed (changes requested):
LGTM, just a couple extra things I think need to be changed.
- Difference hashes are in a different format, please use the same format as the previous commits.
- Please split up bcachefs-tools and bcachefs kernel into separate commits, linux-testing-bcachefs.nix and linux-kernels.nix should be in the same commit.
Here's a [patch](https://github.com/NixOS/nixpkgs/files/8352158/fixup.patch.txt) you can use.
```diff
diff --git a/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix b/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix
index 63d47dd8a38..e86f6fced56 100644
--- a/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix
+++ b/pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix
@@ -1,9 +1,9 @@
 { lib
 , fetchpatch
 , kernel
-, date ? "2022-03-09"
-, commit ? "2280551cebc1735f74eef75d650dd5e175461657"
-, diffHash ? "1mrrydidbapdq0fs0vpqhs88k6ghdrvmjpk2zi7xlwj7j32h0nwp"
+, date ? "2022-03-21"
+, commit ? "c38b7167aa5f3b1b91dcc93ade57f30e95064590"
+, diffHash ? "04lgwnng7p2rlz9sxn74n22750kh524xwfws3agqs12pcrvfsm0j"
 , kernelPatches # must always be defined in bcachefs' all-packages.nix entry because it's also a top-level attribute supplied by callPackage
 , argsOverride ? {}
 , ...
diff --git a/pkgs/tools/filesystems/bcachefs-tools/default.nix b/pkgs/tools/filesystems/bcachefs-tools/default.nix
index aa049682b83..e5325b1f988 100644
--- a/pkgs/tools/filesystems/bcachefs-tools/default.nix
+++ b/pkgs/tools/filesystems/bcachefs-tools/default.nix
@@ -22,13 +22,13 @@
 
 stdenv.mkDerivation {
   pname = "bcachefs-tools";
-  version = "unstable-2022-03-09";
+  version = "unstable-2022-03-22";
 
   src = fetchFromGitHub {
     owner = "koverstreet";
     repo = "bcachefs-tools";
-    rev = "3e2e3d468eed1d5ebbb4c6309d2eaebd081912c5";
-    sha256 = "1sb0dj2whlp3dxgf642z7yx7s8va5ah82zi6r4qni7l64qy1n554";
+    rev = "f3cdace86c8b60a4efaced23b2d31c16dc610da9";
+    sha256 = "1hg4cjrs4yr0mx3mmm1jls93w1skpq5wzp2dzx9rq4w5il2xmx19";
   };
 
   postPatch = ''
diff --git a/pkgs/top-level/linux-kernels.nix b/pkgs/top-level/linux-kernels.nix
index e73b5ebda1d..bd7bacf3000 100644
--- a/pkgs/top-level/linux-kernels.nix
+++ b/pkgs/top-level/linux-kernels.nix
@@ -186,7 +186,7 @@ in {
        else testing;
 
     linux_testing_bcachefs = callPackage ../os-specific/linux/kernel/linux-testing-bcachefs.nix rec {
-      kernel = linux_5_15;
+      kernel = linux_5_16;
       kernelPatches = kernel.kernelPatches;
    };
 
```

Thankfully, the on-disk format change seems to be automatic, so we should be able to backport this as well.

Commit:
linux_testing_bcachefs: 2022-03-09 -> 2022-03-21

This should fix a bug causing kernel panics when your fs has snapshots
enabled. See:
https://lore.kernel.org/all/bc622d24-9fad-7b3-22cb-da4bf2dd32d@ewheeler.net/T/

This patch also bumps the kernel version to 5.16, as bcachefs is devel-
oping against that now.

Commit:
bcachefs-tools: 2022-03-09 -> 2022-03-22

Madouura reviewed (approved):
LGTM

davidak commented:
@DarkKirb thanks for your contribution!

Please follow the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) guidelines for the commit title.

DarkKirb commented:
Reworded both commits

davidak commented:
Result of `nixpkgs-review pr 165763` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_testing_bcachefs.anbox</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.ena</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.ixgbevf</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.mxu11x0</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.ndiswrapper</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_legacy340</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.openafs_1_9</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.phc-intel</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.r8125</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8188eus-aircrack</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8723bs</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8812au</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtlwifi_new</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.sch_cake</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.tbs</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.xmm7360-pci</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_vulkan_beta</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.openafs</li>
  </ul>
</details>
<details>
  <summary>78 packages built:</summary>
  <ul>
    <li>bcachefs-tools</li>
    <li>linuxKernel.kernels.linux_testing_bcachefs</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.acpi_call</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.akvcam</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.apfs</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.asus-ec-sensors</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.asus-wmi-sensors</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.batman_adv</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.bbswitch</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.broadcom_sta</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.can-isotp</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.chipsec</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.cpupower</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.cryptodev</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.ddcci-driver</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.dddvb</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.digimend</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.dpdk</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.dpdk-kmods</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.evdi</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.facetimehd</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.fwts-efi-runtime</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.gcadapter-oc-kmod</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.hid-nintendo</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.hyperv-daemons</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.intel-speed-select</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.isgx</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.it87</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.jool</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.kvmfr</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.liquidtux</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.lttng-modules</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.mba6x_bl</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.mbp2018-bridge-drv</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.mwprocapture</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.netatop</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_beta</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_legacy470</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.nvidiabl</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.openrazer</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.perf</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.ply</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.r8168</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rr-zen_workaround</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8189es</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8192eu</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8814au</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8821au</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8821ce</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl8821cu</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl88x2bu</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.rtl88xxau-aircrack</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.sysdig</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.system76</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.system76-acpi</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.system76-io</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.systemtap</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.tmon</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.tp_smapi</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.turbostat</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.tuxedo-keyboard</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.usbip</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.v4l2loopback</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.v86d</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.veikk-linux-driver</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.vendor-reset</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.vhba</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.virtio_vmmci</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.virtualbox</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.virtualboxGuestAdditions</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.vmm_clock</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.x86_energy_perf_policy</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.xone</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.xpadneo</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.zenpower</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.zfs (linuxKernel.packages.linux_testing_bcachefs.zfsUnstable)</li>
  </ul>
</details>

davidak reviewed (approved):
Change looks good

Builds, tests successful

Madouura commented:
Here's a proper backport. #165790
